src/Mod/PartDesign/Gui/Utils.cpp

```diff
@@ -153,6 +141,85 @@ PartDesign::Body *getBody(bool messageIfNot, bool autoActivate, bool assertModer
     return activeBody;
 }
 
+PartDesign::Body * makeBodyActive(App::DocumentObject* body, App::Document *doc)
+{
+    App::DocumentObject *parent = 0;
+    std::string sub;
+
+    for(auto &v : body->getParents()) {
+        if(v.first->getDocument()!=doc)
+            continue;
+        if(parent) {
+            body = 0;
+            break;
+        }
+        parent = v.first;
+        sub = v.second;
+    }
+
+    if(body) {
+        auto _doc = parent?parent->getDocument():body->getDocument();
+        _FCMD_DOC_CMD(Gui, _doc, "ActiveView.setActiveObject('" << PDBODYKEY
+                      << "'," << Gui::Command::getObjectCmd(parent?parent:body)
+                      << ",'" << sub << "')");
+        // return Gui::Application::Instance->activeView()->
+        //     getActiveObject<PartDesign::Body*>(PDBODYKEY,topParent,subname);
+    }
+
+    return dynamic_cast<PartDesign::Body*>(body);
+}
+
+PartDesign::Body * needActiveBodyMessage (App::Document *doc,
+                                          const QString& infoText)
+{
+    PartDesign::Body* activeBody = nullptr;
+    QDialog dia(Gui::getMainWindow());
+    PartDesignGui::Ui_DlgActiveBody dlg;
+    dlg.setupUi(&dia);
+
+    QObject::connect(dlg.bodySelect, SIGNAL(itemDoubleClicked(QListWidgetItem *)),
+                     &dia, SLOT(accept()));
+
+    if(!infoText.isEmpty()) {
+        dlg.label->setText(infoText + QObject::tr("\n\n") +
+                           QObject::tr("Please select"));
+    }
+
+    auto bodies = doc->getObjectsOfType(PartDesign::Body::getClassTypeId());
+
+    PartDesign::Body* bodyOfActiveObject = nullptr;
+    for (Gui::SelectionSingleton::SelObj obj :  Gui::Selection().getSelection()) {
```

Comment:
I suggest the more idiomatic (and probably slightly more efficient):
```suggestion
    for (const auto &obj :  Gui::Selection().getSelection()) {
```